pkglistgen.py

```diff
@@ -732,21 +733,32 @@ def do_update(self, subcmd, opts):
                 if not os.path.exists(d):
                     os.makedirs(d)
 
+                # Fetch state before mirroring in-case it changes during download.
+                state = repository_state(self.tool.apiurl, project, repo)
```

Comment:
this is buggy - it's pretty non-obvious, but it turns out the repository_state is not the state of the repository's binaries, but just the state of the build statuses. So we will skip package list generation of all packages go from succeeded to succeeded in between 2 checks.

This lead to outdated caches on SP1. I reverted this change locally, but we need a proper fix.

One possibility is always mirroring and then calculating a state hash based on the mirrored directory ourselves. 
@mlschroe suggested to have another view parameter on the repository to get e.g. the mtime of the solv file, so that's another possibility.

Reply:
My first thought was to hash the mirrored file list, but figured repo state hash was better. Hmm, having the extra view parameter would seem better as it would work for repo-checker as well, but if that isn't going to happen can go ahead with hash of file list.